### ISSUE TITLE
Language matching should add likely subtags again after clearing the region

### ISSUE THREAD

**Issue by stasm** [projectfluent/fluent.js#334]
_Originally reported in https://github.com/mozilla-services/screenshots/issues/5347._

After digging into `fluent-langneg` code and comparing it to `LocaleService.cpp`, I think I can confirm that this is a bug in `fluent-langneg`. It comprises three parts.

1. The list of likely subtags in `fluent-langneg` is limited. It currently doesn't include data fro many locales, including `zh-HK`.
2. The matching algorithm in `fluent-langneg` is missing an extra step. After the region of the requested language is set to `*`, `LocaleService.cpp` tries to add likely subtags _again_. In combination with the step in which the requested region is cleared, this allows the algorithm to correctly select `zh-HK` based on the script used.
3. The list of available locales claims to be _unordered_, but in reality it uses its inherent ordering in some of the steps of the algorithm.

In other words, what currently happens is:

1. Request `zh-HK`.
1. Clear the region → `zh-*`.
1. Match against `zh-*-CN-*` and `zh-*-TW-*`.
1. Return `zh-CN` followed by `zh-TW`.

After adding the missing likely subtags, the bug persists:

1. Request `zh-HK`.
1. Add the likely subtag → `zh-Hant-HK`.
1. Clear the region → `zh-Hant-*`.
1. Match against `zh-*-CN-*` and `zh-*-TW-*`.
1. Return `zh-CN` followed by `zh-TW`.

Here's what should happen instead when we add the missing step to the algorithm:

1. Request `zh-HK`.
2. Add the likely subtag → `zh-Hant-HK`.
3. Clear the region → `zh-Hant-*`.
4. Add the likely subtag again → `zh-Hant-TW`.
5. Match against  `zh-*-TW-*`.
6. Return `zh-TW`.

**Comment by zbraniecki** [comment]
taking. Thank you stas!

**Comment by stasm** [comment]
This was fixed two weeks ago in #335. Thanks, @zbraniecki!
